### PR TITLE
Allow Alternate files to be loaded

### PR DESCRIPTION
**Pull request by rmosleydb**
Updated the load_training_dataset function to allow an override to an alternate file, similar to the load_tokenizer function. This will help when fine-tuning or training with different datasets.

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
Seems fine, WDYT @matthayes ?

File: training/trainer.py [hunk 1]
```diff
@@ -85,9 +85,9 @@ def preprocess_batch(batch: Dict[str, List], tokenizer: AutoTokenizer, max_lengt
     )
 
 
-def load_training_dataset() -> Dataset:
-    logger.info(f"Loading dataset from {DATABRICKS_DOLLY_15K_PATH}")
-    dataset = load_dataset("json", data_files=str(DATABRICKS_DOLLY_15K_PATH))["train"]
+def load_training_dataset(data_file_path: str = DATABRICKS_DOLLY_15K_PATH) -> Dataset:
```
Comment (matthayes) [inline training/trainer.py:88]: This looks fine but can you also need to make it possible to override it through the CLI and notebook.  For example, there is a `--input-model` option to change the default model.  There could be a `--training-dataset` option and a combobox in the notebook.  I think you can follow `input_model` as an example to follow.

**Review by matthayes — CHANGES_REQUESTED** [review]

File: training/trainer.py (hunk 1 above)
Comment (srowen) [reply training/trainer.py:88]: I am about to suggest a change right here that will let you specify a general path or dataset ref, b/c we have the dataset on HF Datasets now, too

**Comment by srowen** [comment]
@rmosleydb What about this? https://github.com/databrickslabs/dolly/pull/123

**Comment by rmosleydb** [comment]
Looks good. Thanks.

**Pull request closed without merging**